copilot-chat-request.el

```diff
@@ -77,7 +76,7 @@ If your browser does not open automatically, browse to %s."
                  ("editor-version" . "Neovim/0.10.0")
                  ("user-agent" . "CopilotChat.nvim/2.0.0"))
       :data (format "{\"client_id\":\"Iv1.b507a08c87ecfe98\",\"device_code\":\"%s\",\"grant_type\":\"urn:ietf:params:oauth:grant-type:device_code\"}" device-code)
-      :parser 'json-read
+      :parser (apply-partially 'json-parse-buffer :object-type 'alist :false-object :json-false :null-object nil)
```

Comment:
`json.c` does not use `:json-false`, but `:false` is assigned by default, so there is no need to force a custom setting.

Reply:
No, the original function is `json-read` and this is in the `request`'s parser function so it's detached from the rest of json serialization. It would be better to keep it compatible with the output of `json-read`. If it turns out those extras custom settings are not necessary, it should be fixed in another PR.

Reply:
I see the reasoning behind this.

In my view, the false value only comes into play with o1 requests, but 
Conversely, if that's the case, there's no significant risk in configuring it to maintain compatibility.
If there is any risk, it would be continuing to use settings meant for older packages even when using newer packages.
While I personally don't feel great about continuing with the old values, I'm not the owner of this repository, so I don't have sufficient justification to explicitly request changes.